python-data-version-control/src/evaluate.py

```diff
@@ -0,0 +1,29 @@
+from joblib import load
+import json
+from pathlib import Path
+import numpy as np
+import pandas as pd
+from sklearn.metrics import accuracy_score
+
+from train import load_images, load_labels, preprocess
+
+
+def main(repo_path):
+    test_csv_path = repo_path / "data/prepared/test.csv"
+    test_df = pd.read_csv(test_csv_path)
+    raw_images = load_images(data_frame=test_df, column_name="filename")
+    labels = load_labels(data_frame=test_df, column_name="label")
+    preprocessed = [preprocess(image) for image in raw_images]
+    test_data = np.concatenate(preprocessed, axis=0)
```

Comment:
I'm wondering if it would make sense to refactor this out into a function called `load_data()` that returns a tuple of containing the list of labels and the processed images? Something like:

```python
def load_data(data_path):
    df = pd.read_csv(data_path)
    labels = load_labels(data_frame=df, column_name="label")
    raw_images = load_images(data_fram=df, column_name="filename")
    processed_images = [preprocess(image) for image in raw_images]
    data = np.concatenate(processed_images, axis=0)
    return labels, data
```

Then you could just call the function in `main()`.

That might make it easier to describe what's going on when you do your line-by-line breakdown in the article.

Just a thought!

Reply:
Sounds good, done!